tex/texlive-bin/Portfile

```diff
@@ -289,6 +303,145 @@ post-destroot   {
+    #
+    # The directories MacPorts-TeXLive.texdist and MacPorts-teTeX.texdist in
+    # /Library/TeX/Distributions/ and MacPorts-TeXLive and MacPorts-teTeX in
+    # /Library/TeX/Distributions/.FactoryDefaults/ are considered by
+    # TeX Live / MacTeX developers as owned by MacPorts.
+    #
+    # From a private email exchange with Richard Koch (April 2015):
+    #
+    #    "Do you have direct contact with the MacPorts people. If so, you
+    #     might propose that
+    #     [...]
+    #     b) From now on, they install the data structure themselves.
+    #     Surely b) is the correct option, and I follow it for all
+    #     distributions except Fink-tetex, MacPorts-teTeX, and
+    #     MacPorts-TeXLive."
+    #
+    #    "> I'm willing to help improving this (e.g. I could prepare a port
+    #     > for the data structure in MacPorts), [...]
+    #     AHA. Then it is great that you wrote. Go ahead and write your own
+    #     data, and reuse the names I used, because then MacTeX will not
+    #     overwrite it."
+    #
+    #    "From now on, these entries will not be created by MacTeX. If the
+    #     entries already exist, either because we wrote them or because you
+    #     changed them, MacTeX won't change them. If you remove the entries
+    #     entirely, then new versions of MacTeX won't add them back, but the
+    #     rare use of old install packages will."
+    #
+    # Moreover, they are useful only if MacPorts' TeX Live is installed, so
+    # removing the old ones here and not restoring them in post-deactivate
+    # does no harm to the user. Consider also that teTeX does not exist
+    # anymore in MacPorts.
+
+    xinstall -d ${destroot}${td_mp}
+    ln -s ..${td_fd}/${td_name}/Contents ${destroot}${td_mp}/Contents
+
+    xinstall -d ${destroot}${td_mpfd}/AllTexmf
+    ln -s ${texlive_texmfports} ${destroot}${td_mpfd}/AllTexmf/texmf
+    ln -s ${texlive_texmfdist} ${destroot}${td_mpfd}/AllTexmf/texmf-dist
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/AllTexmf/texmf-doc
+    ln -s ${texlive_texmflocal} ${destroot}${td_mpfd}/AllTexmf/texmf-local
+    ln -s ${texlive_texmfsysvar} ${destroot}${td_mpfd}/AllTexmf/texmf-var
+
+    xinstall -d ${destroot}${td_mpfd}/Doc
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/Doc/texmf-dist-doc
+    ln -s ${texlive_texmfports}/doc ${destroot}${td_mpfd}/Doc/texmf-doc
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/Doc/texmf-doc-doc
+    ln -s ${texlive_texmfdist}/doc ${destroot}${td_mpfd}/Doc/texmf-var-doc
+
+    ln -s ${prefix}/share/info ${destroot}${td_mpfd}/Info
+    ln -s ${prefix}/share/man ${destroot}${td_mpfd}/Man
+
+    # The texdist script determines if a TeX distribution is usable
+    # by testing the binaries in the directory called `uname -p`.
+    # `uname -p` outputs i386 also for x86_64 systems.
+    xinstall -d ${destroot}${td_mpfd}/Programs
+    if {[variant_isset universal]} {
```

Comment:
I would talk to Richard if there's an easier way to do this.

I mean: I'm ok to accept this as is. But it would be nice to simplify it at some point.